test/controllers/register/trustees/individual/CheckYourAnswersIndividualContactDetailsControllerSpec.scala

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers.register.trustees.individual
+
+import controllers.ControllerSpecBase
+import controllers.actions._
+import controllers.behaviours.ControllerAllowChangeBehaviour
+import identifiers.register.trustees.individual.{TrusteeEmailId, TrusteeNameId, TrusteePhoneId}
+import models.Mode.checkMode
+import models._
+import models.person.PersonName
+import models.requests.DataRequest
+import play.api.mvc.{AnyContent, Call}
+import play.api.test.Helpers._
+import utils.checkyouranswers.CheckYourAnswers.StringCYA
+import utils.{AllowChangeHelper, CountryOptions, FakeCountryOptions, FakeDataRequest, UserAnswers}
+import viewmodels.AnswerSection
+import views.html.check_your_answers
+
+class CheckYourAnswersIndividualContactDetailsControllerSpec extends ControllerSpecBase with ControllerAllowChangeBehaviour {
+
+  private val index = Index(0)
+  private val srn = Some("test-srn")
+  private implicit val fakeCountryOptions: CountryOptions = new FakeCountryOptions
+
+  private def submitUrl(mode: Mode = NormalMode, srn: Option[String] = None): Call =
+    routes.CheckYourAnswersIndividualContactDetailsController.onSubmit(mode, index, srn)
+
+  private def answerSection(mode: Mode, srn: Option[String] = None)(implicit request: DataRequest[AnyContent]): Seq[AnswerSection] = {
+    val userAnswers = request.userAnswers
+    val trusteeName = userAnswers.get(TrusteeNameId(index)).map(_.fullName).value
+
+    Seq(AnswerSection(None,
+      StringCYA[TrusteeEmailId](
+        Some(messages("messages__common_email__heading", trusteeName)),
+        Some(messages("messages__common_email__visually_hidden_change_label", trusteeName))
+      )().row(TrusteeEmailId(index))(
+        routes.TrusteeEmailController.onPageLoad(checkMode(mode), Index(index), srn).url, userAnswers) ++
+
+        StringCYA[TrusteePhoneId](
+         Some(messages("messages__common_phone__heading", trusteeName)),
+          Some(messages("messages__common_phone__visually_hidden_change_label", trusteeName))
+        )().row(TrusteePhoneId(index))(
+          routes.TrusteePhoneController.onPageLoad(checkMode(mode), Index(index), srn).url, userAnswers)
+    ))
+  }
+
+  def controller(dataRetrievalAction: DataRetrievalAction = getEmptyData,
```

Comment:
We should probably be using `applicationBuilder` here to make the intent of the code clearer and get rid of the mechanics of creating instances.
https://github.com/hmrc/pensions-scheme-frontend/blob/master/test/base/SpecBase.scala#L60-L69